ui/js/controllers/perf/compare.js

```diff
@@ -772,12 +774,14 @@ perf.controller('CompareSubtestResultsCtrl', [
                                 parent_signature: $scope.newSignature,
                                 framework: $scope.filterOptions.framework,
                             }).then(function (newSeriesList) {
-                                $scope.platformList = [...new Set(_.union(
-                                    $scope.platformList,
-                                    newSeriesList.map(series => series.platform)))];
-                                $scope.testList = [...new Set(_.union(
-                                    $scope.testList,
-                                    newSeriesList.map(series => series.name)))];
+                                $scope.platformList = [...new Set([...new Set([
```

Comment:
Sorry I didn't see this before.  We also don't need double ``new Set``.  A single version of that will do.